lib/spack/llnl/util/tty/log.py

```diff
@@ -89,42 +130,126 @@ def __init__(self, stream):
         """
         self.stream = stream
 
+    def _is_background(self):
+        """True iff """
+        return _is_background_tty(self.stream)
+
+    def _get_canon_echo_flags(self):
+        """Get current termios canonical and echo settings."""
+        cfg = termios.tcgetattr(self.stream)
```

Comment:
I notice at the top of the file we have a try-except for importing `termios`. Do we need to check whether or not this is None first?

Reply:
this is only called from one place, if `self.old_cfg` is set, and that is only set when we have `termios`